pkg/reconciler/instances/connectivityproxy/rendering/chartproviderwithauthentication_test.go

```diff
@@ -0,0 +1,57 @@
+package rendering
+
+import (
+	"fmt"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/chart"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/chart/mocks"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"net/http"
+	"testing"
+)
+
+func TestNewProviderWithAuthentication(t *testing.T) {
+	t.Run("must add authenticator to Component", func(t *testing.T) {
+		// given
+		externalComponentAuthenticatorMock := &mocks.ExternalComponentAuthenticator{}
+
+		chartProviderMock := &mocks.Provider{}
+
+		matcher := func(component *chart.Component) bool {
+			return component.ExternalComponentAuthentication() == externalComponentAuthenticatorMock
+		}
+		chartProviderMock.On("RenderManifest", mock.MatchedBy(matcher)).Return(nil, nil)
+
+		chartProvider := NewProviderWithAuthentication(chartProviderMock, externalComponentAuthenticatorMock)
+
+		// when
+		builder := chart.NewComponentBuilder("1.0.0", "test")
+
+		_, err := chartProvider.RenderManifest(builder.Build())
+		require.NoError(t, err)
+
+		// then
+		chartProviderMock.AssertExpectations(t)
+	})
+}
+
```

Comment:
Suggestion: change function name to `TestExternalComponentAuthenticator_Do `

Reply:
Done